storage/mysql/log_storage.go

```diff
@@ -220,7 +220,7 @@ func (m *mySQLLogStorage) beginInternal(ctx context.Context, treeID int64, reado
 		ctx,
 		m.admin,
 		treeID,
-		trees.GetOpts{TreeType: trillian.TreeType_LOG, Readonly: readonly})
+		trees.NewGetOpts(readonly, trillian.TreeType_LOG))
```

Comment:
The motivation for what I'm doing is to get rid of this from storage. It should have the options passed in. I thought I'd have this done by now but there have been a lot of recent API changes that I have to rework.

Reply:
BTW to be clear I didn't mean you should make that change. It's what I'm working on.

Reply:
Sounds good. Hope you get it done eventually :)

Reply:
If I can stop Al for a bit I might have a chance :-)

The multiple tree types is an additional complication. At the moment if you have a preordered log tree we'll lose that in the call from here. I don't think that will break you at the moment but it would be a problem if the types require different handling for validation etc.

Reply:
```
01001000 01100101 01111001 00101100 00100000 00101010 01110011 01101111 01101101 
01100101 01101111 01101110 01100101 00101010 00100000 01110011 01110100 01101111 
01101100 01100101 00100000 01101101 01111001 00100000 01101011 01100101 01111001 
01100010 01101111 01100001 01110010 01100100 00101100 00100000 01001001 00100111 
01101101 00100000 01110111 01110010 01101001 01110100 01101001 01101110 01100111 
00100000 01110100 01101000 01101001 01110011 00100000 01100010 01111001 00100000 
01110011 01101000 01101111 01110010 01110100 01101001 01101110 01100111 00100000 
01101111 01110101 01110100 00100000 01110100 01101000 01100101 00100000 01110111 
01101001 01110010 01100101 01110011 00101110 
```

Reply:
Is that a puzzle?

Reply:
hehe

Reply:
Xrlobneqf pna or irel sentvyr. Lbh fubhyq or pnershy va pnfr fbzr fbeg nppvqrag unccraf gb gur arj bar.

Reply:
13